### PR TITLE
[jmxfetch] properly stop on windows :bug:

### PR DESCRIPTION
**Pull request by yannmh**
Enhance `ProcessWatchDog` win32/agent.py class to support 'cleanup'
methods. When set, a thread is attached to the process, waits for a
trigger and invoke 'cleanup'.

Alternative here: https://github.com/DataDog/dd-agent/pull/1643

### CODE REVIEW

File: win32/agent.py [hunk 1]
```diff
@@ -5,7 +5,6 @@
 from optparse import Values
 import servicemanager
 import sys
-import threading
```
Comment (remh) [inline win32/agent.py]: Can you reorganize the imports while you're at it please ?

**Comment by remh** [comment]
If i read the code properly stop() is not called:
https://github.com/DataDog/dd-agent/blob/yann/fix-win-jmx-no-restart/win32/agent.py#L82-L89

So i don't think this fix works.

I think we should use a simpler solution and just send a suicide signal to jmxfetch through a temp file, it would be way easier to read and to implement and will also remove a layer of complexity as you won't need your new thread anymore.

**Comment by yannmh** [comment]
Thanks for your feedback @remh. 
You're right it should be changed to `stop` instead: `ProcessWatchDog ` will then trigger `terminate` or a custom `cleanup` method depending on the situation.
https://github.com/DataDog/dd-agent/blob/yann/fix-win-jmx-no-restart/win32/agent.py#L184

I am writing the suggested alternative, i.e. communicating exit order through a file, so we can compare both solutions. I'll let you know when the PR is ready.

**Comment by remh** [comment]
closing in favor of https://github.com/DataDog/dd-agent/pull/1643

**Pull request closed without merging**